libcudacxx/include/cuda/std/__atomic/functions/cuda_ptx_generated_helper.h

```diff
@@ -110,61 +111,55 @@ struct __atomic_longlong2
+                                                                                                          // atom.add.s64
+      __type_switch<size_t,
+                    sizeof(_Type),
+                    __type_case<size_t, 1, __atomic_cuda_operand_deduction<uint8_t, __atomic_cuda_operand_u8>>,
+                    __type_case<size_t, 2, __atomic_cuda_operand_deduction<uint16_t, __atomic_cuda_operand_u16>>,
+                    __type_case<size_t, 4, __atomic_cuda_operand_deduction<uint32_t, __atomic_cuda_operand_u32>>,
+                    __type_default<__atomic_cuda_operand_deduction<uint64_t, __atomic_cuda_operand_u64>>>>>;
 
 template <class _Type>
-using __atomic_cuda_deduce_minmax =
-  _If<_CCCL_TRAIT(is_floating_point, _Type),
-      _If<sizeof(_Type) == 4,
-          __atomic_cuda_operand_deduction<float, __atomic_cuda_operand_f32>,
-          __atomic_cuda_operand_deduction<double, __atomic_cuda_operand_f64>>,
-      _If<_CCCL_TRAIT(is_signed, _Type),
-          _If<sizeof(_Type) == 1,
-              __atomic_cuda_operand_deduction<int8_t, __atomic_cuda_operand_s8>,
-              _If<sizeof(_Type) == 2,
-                  __atomic_cuda_operand_deduction<int16_t, __atomic_cuda_operand_s16>,
-                  _If<sizeof(_Type) == 4,
-                      __atomic_cuda_operand_deduction<int32_t, __atomic_cuda_operand_s32>,
-                      __atomic_cuda_operand_deduction<int64_t, __atomic_cuda_operand_s64>>>>, // atom.min|max.s64
-                                                                                              // supported
-          _If<sizeof(_Type) == 1,
-              __atomic_cuda_operand_deduction<uint8_t, __atomic_cuda_operand_u8>,
-              _If<sizeof(_Type) == 2,
-                  __atomic_cuda_operand_deduction<uint16_t, __atomic_cuda_operand_u16>,
-                  _If<sizeof(_Type) == 4,
-                      __atomic_cuda_operand_deduction<uint32_t, __atomic_cuda_operand_u32>,
-                      __atomic_cuda_operand_deduction<uint64_t, __atomic_cuda_operand_u64>>>>>>;
+using __atomic_cuda_deduce_minmax = _If<
+  _CCCL_TRAIT(is_floating_point, _Type),
+  _If<sizeof(_Type) == 4,
+      __atomic_cuda_operand_deduction<float, __atomic_cuda_operand_f32>,
+      __atomic_cuda_operand_deduction<double, __atomic_cuda_operand_f64>>,
+  _If<_CCCL_TRAIT(is_signed, _Type),
+      __type_switch<size_t,
+                    sizeof(_Type),
+                    __type_case<size_t, 1, __atomic_cuda_operand_deduction<int8_t, __atomic_cuda_operand_s8>>,
+                    __type_case<size_t, 2, __atomic_cuda_operand_deduction<int16_t, __atomic_cuda_operand_s16>>,
+                    __type_case<size_t, 4, __atomic_cuda_operand_deduction<int32_t, __atomic_cuda_operand_s32>>,
+                    __type_default<__atomic_cuda_operand_deduction<int64_t, __atomic_cuda_operand_s64>>>, // atom.min|max.s64
```

Comment:
So far as I can see this could be
```suggestion
      __type_switch<sizeof(_Type),
                    __type_case<1, __atomic_cuda_operand_deduction<int8_t, __atomic_cuda_operand_s8>>,
                    __type_case<2, __atomic_cuda_operand_deduction<int16_t, __atomic_cuda_operand_s16>>,
                    __type_case<4, __atomic_cuda_operand_deduction<int32_t, __atomic_cuda_operand_s32>>,
                    __type_default<__atomic_cuda_operand_deduction<int64_t, __atomic_cuda_operand_s64>>>, // atom.min|max.s64
```

Without any loss of functionality because we always require an integral constant anyway

Reply:
I was thinking it could be useful to type-switch over an enum class. But I will take your suggestion anyway.

Reply:
I guess it might be but it feels like it would be better to then cast to the underlying type